torch/fx/_symbolic_trace.py

```diff
@@ -430,9 +489,29 @@ def call_module(
             value was returned from the ``Module`` invocation.
         """
         module_qualified_name = self.path_of_module(m)
-        if not self.is_leaf_module(m, module_qualified_name):
-            return forward(*args, **kwargs)
-        return self.create_proxy("call_module", module_qualified_name, args, kwargs)
+        with ScopeContextManager(self.scope, m, module_qualified_name):
+            if not self.is_leaf_module(m, module_qualified_name):
+                ret_val = forward(*args, **kwargs)
+            else:
+                ret_val = self.create_proxy("call_module", module_qualified_name, args, kwargs)
+        return ret_val
+
+    def create_node(
+        self,
+        kind: str,
+        target: Target,
+        args: Tuple[Argument, ...],
+        kwargs: Dict[str, Argument],
+        name: Optional[str] = None,
+        type_expr: Optional[Any] = None
+    ) -> Node:
+        node = super().create_node(kind, target, args, kwargs, name, type_expr)
+        self.node_name_to_scope[node.name] = (
+            self.scope.module_path,
+            self.scope.module_type,
+        )
+        node.meta['nn_module_stack'] = {self.scope.module_path : self.scope.module_type}
```

Comment:
This is different from what is store in nn_module_stack from the dynamo trace... 
Ideally, we want to make it consistent... 

node.meta['nn_module_stack'] is currently an OrderedDictionary that store the entire stack of modules. 

@jerryzh168, do you have a preference which structure it should take?

Reply:
I think today we actually only use a single level of scope (modue_fqn, module_type) in quantization (e.g. ("sub.conv", nn.Conv2d)) (actually the scope context code in this PR is the same as the ones defined in quantization tracer: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/tracer.py#L11), but I think theoretically we should be storing a stack of (module_fqn, module_type) 
e.g. 
for a module like following:
class Sub(...):
    def __init__(self, ...):
          self.conv = nn.Conv2d(...)

class M(...):
    def __init__(self, ...):
          self.sub = Sub()

     def forward(self, x):
           return self.sub(x)

for the conv aten op, the scope should be something like
[("sub.conv", nn.Conv2d), ("sub", Sub)] since to be precise, the conv aten node is in the scope of the Conv module and Sub module. We can get module fqn for sub just by looking at "sub.conv", but we can't get the module type of "Sub" easily if we just have a single entry of ("sub.conv", nn.Conv2d)

Reply:
FYI - I actually just moved the Scope related code from quantization tracer into the base class Tracer so everyone has access to the module scope. 

Doesn't the module_fqn already account for the hierarchy?  Eg when the aten::conv is actually called the module_fqn would be sub.conv and the op would be aten::conv2d.  Ok now I see the issue, if you want to preserve the module type at each level of the hierarchy then you do need a listing of each level of hierarchy.

Reply:
@SherlockNoMad  I just added an ordered dict for module stack.  Now the code writes an ordered dict to node.meta['nn_module_stack']

For the example code above w/ Module and Sub module I get this output  `Node sub_mod_conv_0 - mod {'sub_mod': "<class '__main__.SubMod'>", 'sub_mod.conv_0': "<class 'torch.nn.modules.conv.Conv2d'>"}`